fibers.js

```diff
@@ -0,0 +1,25 @@
+const Future = require('fibers/future');
+const fs = Future.wrap(require('fs'));
+const prependDir = (file) => require('path').join(process.argv[2], file)
+
+const main = () => {
+  Future.task(function() {
+    try {
+      process.stdout.write(
+        fs
+        .readFileFuture(prependDir('index.txt'))
+        .wait()
+        .toString()
+        .replace(/\n$/,'')
+        .split('\n')
+        .map((x) => fs.readFileFuture(prependDir(x)).wait().toString())
+        .join(''))
```

Comment:
Since the file is named __fibers.js__ I don't mind including this approach.

I'd love to see people abandon these complex, lawless approaches and use something like [data.task][1] or [Fluture][2] instead.


[1]: https://github.com/folktale/data.task
[2]: https://github.com/Avaq/Fluture